interfaces/builtin/x11.go

```diff
@@ -60,14 +137,52 @@ socket AF_NETLINK - NETLINK_KOBJECT_UEVENT
 bind
 `
 
+type x11Interface struct {
+	commonInterface
+}
+
+func (iface *x11Interface) AppArmorConnectedSlot(spec *apparmor.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	if !release.OnClassic {
+		old := "###PLUG_SECURITY_TAGS###"
+		new := plugAppLabelExpr(plug)
+		snippet := strings.Replace(x11ConnectedSlotAppArmor, old, new, -1)
+		spec.AddSnippet(snippet)
+	}
+	return nil
+}
+
+func (iface *x11Interface) SecCompPermanentSlot(spec *seccomp.Specification, slot *snap.SlotInfo) error {
```

Comment:
Nitpick: is it easier to this other way around, move all the if `!release.OnClassic` tests to init() below, then if we're on classic we register a interface that has less snippets. If we are on core we hand out the snippets and the udev rules and everything. This should cut all the custom code from this type. It could become a common interface.

Reply:
So yes I can do this (https://pastebin.ubuntu.com/p/Br3GF43cVx/) and it does clean up the syntax a bit.

I am struggling with testing it though, as now the init() depends on release.OnClassic and I'm failing to see where in my test setup I can specify the mock value for that at the right time. This is what I have so far: https://pastebin.ubuntu.com/p/tGSGt3w6tW/
If you've a suggestion I'd appreciate it